### ISSUE TITLE
Incorrect position in OpenMW (OpenGL)

### ISSUE THREAD

**Issue by rharish101** [flightlessmango/MangoHud#81]
My MangoHud config is the default one, with the position set to top-right as the only change. When I open OpenMW, which uses OpenGL, MangoHud is shown in the middle-left as opposed to top-right.

Screenshot:
![MangoHud OpenMW](https://user-images.githubusercontent.com/25344287/76833209-d6320300-6822-11ea-982c-c335cb01def9.png)

My setup:
* OS: Arch Linux
* MangoHud version: 0.3.0 (from the AUR)

**Comment by flightlessmango** [comment]
The underlying problem seems to be that openmw is not giving the correct `GL_VIEWPORT`, it's always set to 800x600. I think this is something that should be fixed in openmw

**Comment by rharish101** [comment]
Thanks, I'll raise the issue there.

**Comment by AnyOldName3** [comment]
It's not always set to 800x600, just when the buffers are swapped. We're not explicitly doing this, OpenSceneGraph is. More details here: https://gitlab.com/OpenMW/openmw/-/issues/5312#note_315016884

**Comment by AnyOldName3** [comment]
I'm bumping this again as I've been tidying OpenMW's issue tracker. From our perspective, either MangoHud is incorrect to assume the full viewport is set right before the buffers are swapped, or OpenSceneGraph is incorrect to set a default viewport size right before swapping buffers. This is something that needs to be discussed between the MangoHud developers and OpenSceneGraph developers, as the problem will exist for every OSG application, so there's not any particular need for the OpenMW team to get involved.

To recap, OpenSceneGraph resets the OpenGL state to what it views as a reasonable default at the end of every frame. That includes setting an 800x600 viewport, regardless of what the default framebuffer's size actually is.

**Comment by flightlessmango** [comment]
It doesn't appear to be an issue anymore, most likely fixed on our end a while back

**Comment by AnyOldName3** [comment]
Cooleo. I'll close the issue on our tracker.

**Comment by rharish101** [comment]
Yes, sorry, this was fixed something back.
